app/models/blueprint.rb

```diff
@@ -8,6 +10,9 @@ class Blueprint < ApplicationRecord
 
   acts_as_miq_taggable
 
+  after_create :set_original_blueprint
```

Comment:
Not sure I like after_create.  This forces two DB hits, and before_create would be preferable, though I see you need it because of the chicken-and-egg with respect to the new id.